client/src/App.js

```diff
@@ -71,6 +87,39 @@ function App() {
     setShelves([...shelves, shelf]);
   }
 
+  function addRefToBookAndShelf(location, bookToUpdate) {
+    updateBook('shelfLocation', location, bookToUpdate);
+    updateBooksInCompartment('storedBooks', location, bookToUpdate);
+  }
+
+  function getBookLocation(book) {
+    if (book.shelfLocation) {
+      const shelf = shelves.find(
+        (shelf) => shelf.id === book.shelfLocation.bookshelf
```

Comment:
See comment above. `bookshelfId` might be easier to understand. The same could be applied to column and compartment.